### ISSUE TITLE
dedrm plugin for calibre does not work

### ISSUE THREAD

**Issue by Ketboga** [apprenticeharper/DeDRM_tools#226]
hi, my calibre does not remove drm and then I followed your instructions and started calibre on debug mode, here is the report.

calibre Debug log
calibre 2.80  embedded-python: True is64bit: False
Windows-8-6.2.9200 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '8', '6.2.9200')
Python 2.7.9
Windows: ('8', '6.2.9200', '', 'Multiprocessor Free')
Successfully initialized third party plugins: DeDRM (6, 5, 3)
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 100.861627907 x 100.55257732
Starting up...
Started up in 8.49 seconds with 0 books
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['\xcd\x92\xce\xfcD\xc2\xc3\x105\xbf\x92\xef\xdeW8\x1e\xa1N+\xe1\x0cU\xc1L\xe3J\xea\xea\xd68\xd4k', 'D:\\Serdar\\Downloads\\Empire of Difference- The Ottomans in Comparative Perspective.azw3']
DeDRM v6.5.3: Trying to decrypt Empire of Difference- The Ottomans in Comparative Perspective.azw3
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Empire of Difference: The Ottomans in Comparative Perspective
Found 4 keys to try after 0.3 seconds
Crypto Type is: 2
DeDRM v6.5.3: Failed to decrypt with error: No key found in 4 keys tried.
DeDRM v6.5.3: Looking for new default Kindle Key after 0.3 seconds
searching for kinfoFiles in C:\Users\serda\AppData\Local
Found K4PC 1.9+ kinf2011 file: C:\Users\serda\AppData\Local\Amazon\Kindle\storage\.kinf2011
Decrypted key file using IDString '3703314478' and UserName '7365726461'
DeDRM v6.5.3: Found 1 new key
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: Empire of Difference: The Ottomans in Comparative Perspective
Found 4 keys to try after 0.5 seconds
Crypto Type is: 2
DeDRM v6.5.3: Ultimately failed to decrypt after 0.5 seconds.
Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 610, in run
  File "calibre_plugins.dedrm.__init__", line 560, in KindleMobiDecrypt
DeDRMError: DeDRM v6.5.3: Ultimately failed to decrypt after 0.5 seconds
Added Empire of Difference: The Ottomans in Comparative Perspective to db in: 2.7
Added 1 books in 4.9 seconds

**Comment by ElleKayEm** [comment]
If that book is from or for an E-ink Kindle (not a Fire), then you need to enter the Kindle's serial number into the plugin customization.  Otherwise, make sure Kindle for PC is version 1.17 or earlier and import your book from there.

**Comment by BoatrightTBC** [comment]
To provide the tiniest bit more info, this is a known problem due to Amazon changing the way they store and encrypt files on the pc and mac kindle apps.  There is an active discussion of how to get around this, but to date, there has been little success.  The suggestion above is the only current work-around.

**Comment by ElleKayEm** [comment]
@BoatrightTBC 

Yes, the 1.19 update to Kindle for PC/Mac allowed it to support the KFX format which cannot be decrypted yet.  Not every book is available in this format -- although most newer ones are.  This file is in the old KF8 (AZW3) format.  It looks like it was downloaded from the website --- another way to get a format that can be decrypted provided you have an E-ink Kindle.

**Comment by Moonshiney** [comment]
I was about to ask for help about several Kindle titles which could not be de-DRMed when I found this discussion. 
My Kindle-For-PC was set to update automatically and V1.19 installed a few days ago. 
After reading this thread, I reverted to V 1.17 and disabled the auto-update before trying to import the problematic titles.
Problem solved! All the problematic titles have come through Calibre as expected.
Thank you to ElleKayEm and BoatrightTBC for explaining.
